### PR TITLE
Update to DG sections 2.8 and 2.9 and Filter function + Clean up DG

### PR DESCRIPTION
**Pull request by YurouTang**
section 2.8 Model
section 2.9 Storage
Filter
+ Overall edit of DG

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -246,39 +246,54 @@ image::DeletePersonSdForLogic.png[width="800"]
...
 As a more OOP model, we can store a `Tag` list in `Car Park Finder`, which `Carpark` can reference. This would allow `Car Park Finder` to only require one `Tag` object per unique `Tag`, instead of each `Carpark` needing their own `Tag` object. An example of how such a model may look like is given below. +
  +
 image:ModelClassBetterOopDiagram.png[width="800"]
+Use of Tag object
 
 [[Design-Storage]]
 === Storage component
 
-The `Storage` component consist of a `StorageManager` that takes care of all the Input/Output actions.
+The Storage component, managed by the `StorageManager`, serves as a backend storage for data of Car Park Finder.
```
Comment (ZhuanYu) [inline docs/DeveloperGuide.adoc]: Please follow monospace formatting

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -246,39 +246,54 @@ image::DeletePersonSdForLogic.png[width="800"]
...
 
 .Structure of the Storage Component
 image::StorageClassDiagram.png[width="800"]
 
 *API* : link:{repoURL}/src/main/java/seedu/address/storage/Storage.java[`Storage.java`]
 
-The `Storage` component,
+The Storage component can perform the following functions:
```
Comment (ZhuanYu) [inline docs/DeveloperGuide.adoc]: Please follow monospace formatting

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -246,39 +246,54 @@ image::DeletePersonSdForLogic.png[width="800"]
...
+
+* The Model component exposes an unmodifiable `ObservableList<Carpark>` that can be 'observed'
+e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list changes.
+
 
 [NOTE]
 As a more OOP model, we can store a `Tag` list in `Car Park Finder`, which `Carpark` can reference. This would allow `Car Park Finder` to only require one `Tag` object per unique `Tag`, instead of each `Carpark` needing their own `Tag` object. An example of how such a model may look like is given below. +
  +
 image:ModelClassBetterOopDiagram.png[width="800"]
+Use of Tag object
```
Comment (ZhuanYu) [inline docs/DeveloperGuide.adoc]: I can't really see the doc properly but why is there a stray text here?

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -441,6 +456,7 @@ image::actdiagram-find.png[width="800", align="left"]
 ** Pros: Less complicated predicate.
 ** Cons: Manipulates the data, editing it for every other case.
 
+
```
Comment (ZhuanYu) [inline docs/DeveloperGuide.adoc]: Extra spacing?

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -609,20 +625,24 @@ filter the car parks accordingly. The various predicates include
 Step 5. Next, `model` updates a filtered list of car parks by checking each car park
 against the specified predicate.
 
-Step 6. The following activity diagram summarizes what happens when a user executes a
+.How the filter operation works
+image::SequenceDiagram-filter.png[width="700", align="left"]
```
Comment (ZhuanYu) [inline docs/DeveloperGuide.adoc]: Is there a mention to the sequence diagram?

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (YurouTang) [reply docs/DeveloperGuide.adoc]: Oh this one is supposed to be a caption to the diagram, but for some reason the font and colour isnt the same as the other captions although the same notation is used

File: docs/DeveloperGuide.adoc (hunk 5 above)
Comment (YurouTang) [reply docs/DeveloperGuide.adoc]: I felt that the steps actually describes the sequence diagram in this case, what do you think?

Comment (ZhuanYu) [reply docs/DeveloperGuide.adoc]: I think we need to have reference to diagrams as per CS2101 style. @Lexzt any thoughts?

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (ZhuanYu) [reply docs/DeveloperGuide.adoc]: This isn't fix still

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (ZhuanYu) [reply docs/DeveloperGuide.adoc]: If its a caption then follow the style of the rest. Use a full stop and place the caption above the picture.

**Comment by ZhuanYu** [comment]
looks good

**Pull request merged**